mlir/test/Transforms/test-legalizer-no-materializations.mlir

```diff
@@ -0,0 +1,67 @@
+// RUN: mlir-opt -allow-unregistered-dialect -split-input-file -test-legalize-patterns="allow-pattern-rollback=0 build-materializations=0 attach-debug-materialization-kind=1" -verify-diagnostics %s | FileCheck %s --check-prefix=CHECK-KIND
```

Comment:
Splitting up the legalization test looks unrelated, do we have to do it as a part of this change?